charts/argo-workflows/templates/controller/workflow-controller-servicemonitor.yaml

```diff
@@ -1,5 +1,7 @@
+{{- $prometheusApiVersion := default "monitoring.coreos.com/v1" .Values.controller.serviceMonitor.apiVersion -}}
```

Comment:
You already provide the default in values.yaml, so it's redundant here.

Reply:
Addressed in 1318f30.

- Included a check to prevent setting the apiVersion to empty.

Reply:
`.Values.controller.serviceMonitor.apiVersion` has default value, so I wonder we need `required` validation. 🤔

Reply:
The validation is just defensive programming in caseYou're right. The capability validation won't pass if `apiVersion` is intentionally set to empty.

Remove the `required` validation in 736df43

Reply:
@cmontemuino 
Thank you!
`#{{- end }}` is not intentional in the commit...? 👀